src/smif/model/sos_model.py

```diff
@@ -149,15 +149,29 @@ def add_model(self, model):
         self.models[model.name] = model
 
     def before_model_run(self, data_handle):
-        """Initialise each model (passing in parameter data only)
+        """Initialise each model (passing in parameter data only) and return the job graph
```

Comment:
I think `before_model_run` is the wrong name for a method that returns a job graph. 

I suggest that the graph stuff should move out of `SosModel` entirely - the two methods apart from this one which handle `DiGraph`s are static, so should be quite portable.